cmd/syncTraverser.go

```diff
@@ -29,6 +29,11 @@ import (
 	"sync/atomic"
 )
 
+// TODO: Get rid of highly situational constructors.
```

Comment:
Agreed. To say that it's equivalent to the mistakes of copy might be an exaggeration though LOL. 

Keep in mind that this was also a refactoring effort, and as with all refactoring efforts, it's a step-by-step process.

Reply:
Wasn't necessarily meaning it was _equivalent_, more or less just meant we were going back down a very similar path. Codebase PTSD, if you would.